### PR TITLE
Bugfix/escape columnname

### PR DESCRIPTION
**Pull request by eycho-am**
*Issue #, if available:* [#274](https://github.com/awslabs/deequ/issues/274)

*Description of changes:*

When there is a . in the column name, Spark recognizes that as trying to get the field nested in a column. This is an issue for those testing with common data sets like Iris, resulting in the following error: 
`org.apache.spark.sql.AnalysisException: Column 'sepal.width' does not exist. Did you mean one of the following? [petal.length, petal.width, sepal.length, sepal.width, variety]`

This PR addresses the issue by using helper methods that add and remove backticks "`" in the appropriate instances when going through the data.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala [hunk 1]
```diff
@@ -204,6 +204,7 @@ abstract class StandardScanShareableAnalyzer[S <: DoubleValuedState[_]](
     entity: Entity.Value = Entity.Column)
   extends ScanShareableAnalyzer[S, DoubleMetric] {
 
+  // Thrown here
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala:207]: nit: Remove comment or rephrase it with more details

File: src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala [hunk 2]
```diff
@@ -217,6 +218,7 @@ abstract class StandardScanShareableAnalyzer[S <: DoubleValuedState[_]](
     }
   }
 
+  // Thrown here
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala:221]: nit: Remove comment or rephrase it with more details

File: src/main/scala/com/amazon/deequ/analyzers/runners/AnalysisRunner.scala [hunk 3]
```diff
@@ -299,6 +299,7 @@ object AnalysisRunner {
     frequenciesAndNumRows.numRows -> results
   }
 
+  // Main method that needs looking into
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/analyzers/runners/AnalysisRunner.scala:302]: nit: Remove comment or rephrase it with more details

File: src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala [hunk 4]
```diff
@@ -303,17 +305,34 @@ object Preconditions {
...
+  }
+
+  def escapeColumn(column: String): String = {
+    if (column.contains(".")) {
+      "`" + column + "`"
+    } else {
+      column
+    }
+  }
+
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala:329]: Can we create a utility object which contains these methods? In the future, one wouldn't immediately know that the important methods for escaping/un-escaping are inside Analyzer.scala

File: src/main/scala/com/amazon/deequ/profiles/ColumnProfiler.scala [hunk 5]
```diff
@@ -450,15 +471,13 @@ object ColumnProfiler {
...
-
       castedData = genericStatistics.typeOf(name) match {
         case Integral => castColumn(castedData, name, LongType)
         case Fractional => castColumn(castedData, name, DoubleType)
         case _ => castedData
       }
     }
-
-    castedData
+  castedData
```
Comment (rdsharma26) [inline src/main/scala/com/amazon/deequ/profiles/ColumnProfiler.scala:480]: nit: inconsistent indentation?

**Review by rdsharma26 — COMMENTED** [review]
Great PR to fix a long-standing issue!

**Comment by eycho-am** [comment]
> When you say that Spark recognizes `.` as trying to get the field nested in a column, does that mean Deequ can work with nested fields? Will `.` work to add constraints on nested fields? If so, can we add a test for this behavior as a means of documentation.

I tried creating a test for this but based on this method it looks like Deequ doesn't work with nested columns:
https://github.com/awslabs/deequ/blob/master/src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala#L354

**Review by rdsharma26 — APPROVED** [review]

**Pull request merged**